### ISSUE TITLE
[Interop] Array#swap invocable, but not readable

### ISSUE THREAD

**Issue by fniephaus** [oracle/truffleruby#1816]
I'm getting errors when trying to display the API of an `Array`, because of [Array#swap](https://github.com/oracle/truffleruby/blob/c3845389c4f38e51283b48a1ea3ccdac4962adbf/src/main/ruby/truffleruby/core/array.rb#L1673-L1678). It's listed as part of `getMembers` and can be invoked. But it does not seem to be a *readable* member. I'm not too familiar with the API of `Array`, but I couldn't find `#swap` in the [class documentation](https://ruby-doc.org/core-2.6.5/Array.html), so it may be TruffleRuby-specific.

#### Repro session
```
$ polyglot --shell --jvm
GraalVM MultiLanguage Shell 19.3.0
Copyright (c) 2013-2019, Oracle and/or its affiliates
  Python version 3.7.4
  Ruby version 2.6.2
python> import polyglot
python> x = polyglot.eval(language='ruby', string='[1, 2]')
python> [key for key in dir(x) if not hasattr(x, key)]
['swap'] # EMPTY LIST EXPECTED
python> x.swap(0,1)
1
python> x
[2, 1] # WORKED AS EXPECTED
```

**Comment by eregon** [comment]
Thank you for the bug report, I can reproduce.
`Array#swap` is indeed an internal method, marked as `protected`.
`protected` is between `public` and `private` and that's probably why it appears in one case but not the other.

@pitr-ch Could you take a look at this?

I think there are two things to fix:
* Should `protected` methods be listed in `getMembers()` and `isMemberReadable()`? We need to be consistent.
* We should probably hide `Array#swap`, `private` + `Truffle.privately {}` or better `Truffle::ArrayOperations.swap`.

**Comment by pitr-ch** [comment]
fixed in 25b684ed274b904d1613c6334f97eaa0bec9e111
